src/taipy/_cli/_migrate.py

```diff
@@ -0,0 +1,487 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import json
+import os
+import sqlite3
+from functools import lru_cache
+from typing import Dict, List, Tuple
+
+import pymongo
+
+from taipy.logger._taipy_logger import _TaipyLogger
+
+__logger = _TaipyLogger._get_logger()
+
+
+def __update_parent_ids(entity: Dict, data: Dict) -> Dict:
```

Comment:
Can we add a docstring clarify which Taipy version "parent_id" has been removed by "parent_ids"?

Further step may be splitting the methods that migrate from Taipy 1.0 -> Taipy 2.0 to a dedicated file, Taipy 2.0 -> Taipy 3.0 to a dedicated file, etc.

Reply:
I believe it was on 2.1. Theres no migration from taipy 1.0 from 2.0. Just 2.X to 3.0. If I recall correctly we had a migration from 1.0 to 2.0 in code. @jrobinAV What do you think about migrating entities from 1.0? It shouldnt be too much work and we could add it later as well, to not block the release.

Reply:
I believe we don t need to migrate from 1.0.

Reply:
After discussion, we proposed to add migration from 1.0 in another PR that should not block the 3.0.0 release.